File: easy/javascript/challenge/commons-chars/commonchars.test.js
const longestCommonChars = require("./commonchars.js");

describe('longestCommonChars', () => {
  it('should return an empty string when one of the strings is empty', () => {
    expect(longestCommonSubsequence('', 'test')).toBe('');
    expect(longestCommonSubsequence('test', '')).toBe('');
  });

  it('should return the entire string when the two strings are identical', () => {
    expect(longestCommonSubsequence('test', 'test')).toBe('test');
  });

  it('should return the longest common chars', () => {
    expect(longestCommonSubsequence('ABCDGH', 'AEDFHR')).toBe('ADH');
    expect(longestCommonSubsequence('AGGTAB', 'GXTXAYB')).toBe('GTAB');
    expect(longestCommonSubsequence('ABCD', 'ACDF')).toBe('ACD');
  });
});

File: easy/java/src/test/java/es/ing/previous/CommonCharsTest.java
package es.ing.previous;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class CommonCharsTest {

    @Test
    public void shouldReturnAnEmptyStringWhenOneOfTheStringsIsEmpty() {
        Assertions.assertEquals(new CommonChars().longestCommonChars("", "test"), "");
        assertEquals(new CommonChars().longestCommonChars("test", ""), "");
    }

    @Test
    public void shouldReturnTheEntireStringWhenTheTwoStringsAreIdentical() {
        assertEquals(new CommonChars().longestCommonChars("test", "test"), "test");
    }


    @Test
    public void shouldReturnTheLongestCommonChars() {
        assertEquals(new CommonChars().longestCommonChars("ABCDGH", "AEDFHR"), "ADH");
        assertEquals(new CommonChars().longestCommonChars("AGGTAB", "GXTXAYB"), "GTAB");
        assertEquals(new CommonChars().longestCommonChars("ABCD", "ACDF"), "ACD");
    }
}